tests/ui/associated-inherent-types/private-in-public.rs

```diff
@@ -2,22 +2,14 @@
 #![feature(type_privacy_lints)]
```

Comment:
The old logic can only be removed when the new lints are not feature gated and run by default, so this PR needs to do stabilization for `type_privacy_lints` as well, at least for `private_interfaces` and `private_bounds.

`unnameable_types` is sort of orthogonal and can be stabilized later and separately, no need to touch it in this PR in any way.